src/libraries/Microsoft.Extensions.Configuration.Xml/src/XmlConfigurationElement.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.Xml
+{
+    internal class XmlConfigurationElement
+    {
+        public string ElementName { get; }
+
+        public string Name { get; }
+
+        public string LineInfo { get; }
+
+        /// <summary>
+        /// The children of this element
+        /// </summary>
+        public List<XmlConfigurationElement>? Children { get; set; }
+
+        /// <summary>
+        /// The siblings of this element, including itself
+        /// Elements are considered siblings if they share the same element name and name attribute
+        /// This list is shared by each sibling
+        /// </summary>
+        public List<XmlConfigurationElement>? Siblings { get; set; }
+
+        public XmlConfigurationElementTextContent? TextContent { get; set; }
+
+        public List<XmlConfigurationElementAttributeValue>? Attributes { get; set; }
+
+        public XmlConfigurationElement(string elementName, string name, string lineInfo)
+        {
+            ElementName = elementName ?? throw new ArgumentNullException(nameof(elementName));
+            Name = name;
+            LineInfo = lineInfo;
+            Children = null;
+            Siblings = null;
+            TextContent = null;
+            Attributes = null;
+        }
+
+        public bool IsSiblingOf(XmlConfigurationElement xmlConfigurationElement)
+        {
+            if (xmlConfigurationElement is null)
+            {
+                throw new ArgumentNullException(nameof(xmlConfigurationElement));
+            }
+
+            return string.Equals(ElementName, xmlConfigurationElement.ElementName)
```

Comment:
Should this comparison be ordinal or culture aware?

Reply:
I think ordinal is the correct choice here. For example, the German 'ß' character is equal to 'ss' when comparing with invariant, but not when comparing with ordinal. And 'ß' is definitely not always the same as 'ss'. 
(I'm Belgian, but I've studied German language)

According to https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0, configuration keys are case insensitive. I'm thinking that should also mean siblings should be compare case insensitively, since they would end up as the same keys eventually anyway. 

I'll update the calls.